### DISCUSSION TITLE
iCloud Private Relay IP is recognised as different country

### DISCUSSION THREAD

**Discussion by Vadimkin** [plausible/analytics#1611] [Self-Hosted Support] [answered]
I am using iCloud Private Relay, my IP is [`104.28.131.167`](https://db-ip.com/104.28.131.167). By db-ip, it is Ukraine (and it's my current location).

Self-hosted Plausible recognizes it as Azerbaijan country, which is not Ukraine.
![image](https://user-images.githubusercontent.com/2240404/149632117-1d33b01c-a29e-4560-93d2-84852d9a63cb.png)

When the last octet changed to `0`, then I see IP from Azerbaijan: https://db-ip.com/104.28.131.0. I am not sure is it related somehow with IP anonymization (as I understand, there is no octets anonymization)

I tried to reproduce this issue on https://plausible.io/plausible.io. I see my visit on the live map, but no data for my country at all.
![image](https://user-images.githubusercontent.com/2240404/149632218-27c47548-44c6-4c2b-9dec-6c119e07e5ac.png)
![image](https://user-images.githubusercontent.com/2240404/149632223-8c118fac-4e8a-4cbf-bf8f-991d9bfffa9d.png)

When iCloud Private Relay is disabled, I see correct data (based on my real IP) on cloud+self-hosted solutions.

The expected behaviour is: User is analytics table located in Ukraine, not Azerbaijan

Environment:
- OS: MacOs 12.2 Beta (21D5039d)
- Browser: Safari
- Browser Version: Version 15.3 (17612.4.8.1.1)
- Docker Image: `8ad1c92d7e2a` (stable) and `ac802d08785b` (master)

**Comment by ukutaht** [comment, marked as answer]
The iCloud Private Relay will mask your real IP address like a VPN. There's nothing we can do on our end to get the correct country in that case.

The best we could try to do is ignore the country if the request comes from a known datacenter/proxy IP address.
